### PR TITLE
Implement translation of Snowflake QUALIFY clause

### PR DESCRIPTION
**Pull request by vil1**
Along with a few code improvements ;).

### CODE REVIEW

File: core/src/main/scala/com/databricks/labs/remorph/parsers/snowflake/SnowflakeExpressionBuilder.scala [hunk 1]
```diff
@@ -6,40 +6,22 @@ import com.databricks.labs.remorph.parsers.snowflake.SnowflakeParser._
...
-      }
-    } else if (ctx.expression_elem() != null) {
-      ctx.expression_elem().accept(this)
-    } else {
-      null
+  override def visitSelect_list_elem(ctx: Select_list_elemContext): ir.Expression = {
+    val rawExpression = ctx match {
+      case c if c.column_elem() != null => c.column_elem().accept(this)
+      case c if c.expression_elem() != null => c.expression_elem().accept(this)
+      case _ => null
```
Comment (nfx) [inline core/src/main/scala/com/databricks/labs/remorph/parsers/snowflake/SnowflakeExpressionBuilder.scala:13]: Bug: "input: ir.Expression" becomes null, which it shouldn't. I guess you have to use Option here

**Review by nfx — CHANGES_REQUESTED** [review]
One small bug

File: core/src/main/scala/com/databricks/labs/remorph/parsers/snowflake/SnowflakeExpressionBuilder.scala (hunk 1 above)
Comment (vil1) [reply core/src/main/scala/com/databricks/labs/remorph/parsers/snowflake/SnowflakeExpressionBuilder.scala:13]: It's actually a not-yet-implemented case, let me just implement it properly. Stay tuned

**Review by nfx — APPROVED** [review]
thank you!

**Pull request merged**